core/services/eth/pool.go

```diff
@@ -43,9 +62,14 @@ func (p *Pool) Dial(ctx context.Context) (err error) {
 	return p.verifyChainIDs(ctx)
 }
 
+// verifyChainIDs checks that every node's chain ID is consistent, initializing from the first node if nil.
 func (p *Pool) verifyChainIDs(ctx context.Context) (err error) {
 	if p.chainID == nil {
-		return nil
+		chainID, err2 := p.nodes[0].ChainID(ctx)
+		if err2 != nil {
+			return errors.Wrap(err, "failed to get chain ID from first node")
+		}
+		p.initChainID(chainID)
```

Comment:
I'm not sure if this bypass was an oversight or maybe a testing convenience, but now verification is always executed and some test helpers were adjusted to handle `eth_chainId` automatically.

Reply:
Testing convenience. Mandating the check is the right move.